### PR TITLE
rm `adjoint` & `transpose` adjoints

### PR DESCRIPTION
**Pull request by mcabbott**
closes https://github.com/FluxML/Zygote.jl/issues/1257

### CODE REVIEW

File: src/lib/array.jl [hunk 1]
```diff
@@ -337,34 +337,6 @@ end
...
-end
-
-@adjoint function LinearAlgebra.Adjoint(x)
-  back(Δ) = (LinearAlgebra.Adjoint(Δ),)
-  back(Δ::NamedTuple{(:parent,)}) = (Δ.parent,)
-  return LinearAlgebra.Adjoint(x), back
-end
-
-@adjoint parent(x::LinearAlgebra.Adjoint) = parent(x), ȳ -> (LinearAlgebra.Adjoint(ȳ),)
-@adjoint parent(x::LinearAlgebra.Transpose) = parent(x), ȳ -> (LinearAlgebra.Transpose(ȳ),)
```
Comment (mcabbott) [inline src/lib/array.jl:366]: I forgot that  https://github.com/JuliaDiff/ChainRulesCore.jl/pull/446  isn't merged, so removing these lines probably isn't a good idea. 

Before:
```
julia> gradient(x -> first(parent(x)), [1,2,3]')[1]
1×3 adjoint(::Vector{Float64}) with eltype Float64:
 1.0  0.0  0.0
```
This PR, with tagged CRC:
```
julia> gradient(x -> first(parent(x)), [1,2,3]')[1]
(parent = [1.0, 0.0, 0.0],)
```

Comment (ToucheSir) [reply src/lib/array.jl:366]: One alternative idea is to desguar the `@adjoint` and overload `_pullback` directly. That'll have the benefit of removing some unnecessary layers, while making it easier to do stuff like https://github.com/FluxML/ZygoteRules.jl/issues/22.

Comment (mcabbott) [reply src/lib/array.jl:366]: Glad to see there were tests for this, at least: fails on `isapprox(::NamedTuple{(:parent,), ...}, ::Matrix{Float64})`.

**Comment by mcabbott** [comment]
Ok to merge?

**Review by ToucheSir — APPROVED** [review]
Yes, I hadn't noticed https://github.com/FluxML/Zygote.jl/pull/1259/commits/26ae24b9cb104428d7e69e4717810a6917dc15e6

**Pull request merged**